eth/db/header.py

```diff
@@ -146,46 +146,55 @@ def header_exists(self, block_hash: Hash32) -> bool:
         validate_word(block_hash, title="Block Hash")
         return block_hash in self.db
 
-    def persist_header(self,
-                       header: BlockHeader
-                       ) -> Tuple[Tuple[BlockHeader, ...], Tuple[BlockHeader, ...]]:
+    def persist_header_chain(self,
+                             *headers: BlockHeader
+                             ) -> Tuple[Tuple[BlockHeader, ...], Tuple[BlockHeader, ...]]:
         """
-        :returns: iterable of headers newly on the canonical chain
+        Return two iterable of headers, the first containing the new canonical headers,
+        the second containing the old canonical headers
         """
-        if header.parent_hash != GENESIS_PARENT_HASH:
-            try:
-                self.get_block_header_by_hash(header.parent_hash)
-            except HeaderNotFound:
+
+        previous = None
+
+        for header in headers:
```

Comment:
I think we should check and error out if the headers are not chained. (before starting to do any database writes)

Something like:

```py
consecutive_pairs = sliding_window(2, headers)
if any(parent.hash != child.parent_hash for parent, child in consecutive_pairs):
  raise ValidationError(...)
```

Reply:
I read his comment as an either-or, and the first option is what I suggested. :)

I'm inclined to prefer that not-contiguous headers be added with `for h in headers: persist_header(h)`.

Reply:
:+1: to only allowing contiguous chains and using `for h in headers: persist_header(header)` style for when we need to do non-contiguous chains.

Reply:
>'m inclined to prefer that not-contiguous headers be added with `for h in headers: persist_header(h).`

Yes, I think that makes sense :+1: